beacon-chain/rpc/beacon/committees.go

```diff
@@ -0,0 +1,151 @@
+package beacon
+
+import (
+	"context"
+
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/pagination"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// ListBeaconCommittees for a given epoch.
+//
+// This request may specify validator indices or public keys to filter out
+// validator beacon chain committees. If no filter criteria is specified, the response returns
+// all beacon committees for the current epoch. The results are paginated by default.
+func (bs *Server) ListBeaconCommittees(
+	ctx context.Context,
+	req *ethpb.ListCommitteesRequest,
+) (*ethpb.BeaconCommittees, error) {
+	if int(req.PageSize) > params.BeaconConfig().MaxPageSize {
+		return nil, status.Errorf(
+			codes.InvalidArgument,
+			"requested page size %d can not be greater than max size %d",
+			req.PageSize,
+			params.BeaconConfig().MaxPageSize,
+		)
+	}
+
+	headState := bs.HeadFetcher.HeadState()
+	var requestingGenesis bool
+	var startSlot uint64
+	switch q := req.QueryFilter.(type) {
+	case *ethpb.ListCommitteesRequest_Epoch:
+		startSlot = helpers.StartSlot(q.Epoch)
+	case *ethpb.ListCommitteesRequest_Genesis:
+		requestingGenesis = q.Genesis
+	default:
+		startSlot = headState.Slot
+	}
+
+	var attesterSeed [32]byte
+	var activeIndices []uint64
+	var err error
+	if requestingGenesis || helpers.SlotToEpoch(startSlot) < helpers.SlotToEpoch(headState.Slot) {
+		activeIndices, err = helpers.ActiveValidatorIndices(headState, helpers.SlotToEpoch(startSlot))
+		if err != nil {
+			return nil, status.Errorf(
+				codes.Internal,
+				"could not retrieve active indices for epoch %d: %v",
+				helpers.SlotToEpoch(startSlot),
+				err,
+			)
+		}
+		archivedCommitteeInfo, err := bs.BeaconDB.ArchivedCommitteeInfo(ctx, helpers.SlotToEpoch(startSlot))
+		if err != nil {
+			return nil, status.Errorf(
+				codes.Internal,
+				"could not request archival data for epoch %d: %v",
+				helpers.SlotToEpoch(startSlot),
+				err,
+			)
+		}
+		if archivedCommitteeInfo == nil {
+			return nil, status.Errorf(
+				codes.NotFound,
+				"could not request data for epoch %d, perhaps --archive in the running beacon node is disabled",
+				helpers.SlotToEpoch(startSlot),
+			)
+		}
+		attesterSeed = bytesutil.ToBytes32(archivedCommitteeInfo.AttesterSeed)
+	} else if !requestingGenesis && helpers.SlotToEpoch(startSlot) == helpers.SlotToEpoch(headState.Slot) {
```

Comment:
Otherwise if requesting current epoch, return current results

Reply:
Can you add these as comments? it seems helpful